tutorials/visualization/20_ui_events.py

```diff
@@ -29,72 +29,83 @@
-# to it.
+# The events are modeled after matplotlib's event system. Each event has a string name
+# (the snake-case version of its class name) and a list of relevant values. For example,
+# the "time_change" event should have the new time as a value. Values can be any python
+# object. When publishing an event, the publisher creates a new instance of the event's
+# class. When subscribing to an event, having to dig up and import the correct class is
+# a bit of a hassle. Following matplotlib's example, subscribers use the string name of
+# the event to refer to it.
 #
 # Below, we create a custom plot and then make it publish and subscribe to
-# :class:`~mne.viz.ui_events.TimeChange` events so it can work
-# together with the topomap plots we created earlier.
-fig3, ax = plt.subplots()
-ax.plot(aud_left.times, aud_left.pick("mag").data.max(axis=0), label="Left")
-ax.plot(aud_right.times, aud_right.pick("mag").data.max(axis=0), label="Right")
+# :class:`~mne.viz.ui_events.TimeChange` events so it can work together with the
+# plots we created earlier.
+
+# sphinx_gallery_thumbnail_number = 5
+
+# Recreate the earlier plots
+fig3 = avg_evokeds.plot_topomap("interactive")
+fig4 = stc.plot("sample", subjects_dir=data_path / "subjects")
+
+# Create a custom plot
+fig5, ax = plt.subplots()
+ax.plot(avg_evokeds.times, avg_evokeds.pick("mag").data.max(axis=0))
 time_bar = ax.axvline(0, color="black")  # Our time slider
 ax.set_xlabel("Time (s)")
 ax.set_ylabel("Maximum magnetic field strength")
 ax.set_title("A custom plot")
-plt.legend()
 
 
 def on_motion_notify(mpl_event):
     """Respond to matplotlib's mouse event.
 
-    Publishes an MNE-Python TimeChange event. When the mouse goes out of
-    bounds, the xdata will be None, which is a special case that needs to be
-    handled.
+    Publishes an MNE-Python TimeChange event. When the mouse goes out of bounds, the
+    xdata will be None, which is a special case that needs to be handled.
     """
     if mpl_event.xdata is not None:
-        publish(fig3, TimeChange(time=mpl_event.xdata))
+        publish(fig5, TimeChange(time=mpl_event.xdata))
 
 
 def on_time_change(event):
     """Respond to MNE-Python's TimeChange event. Updates the plot."""
     time_bar.set_xdata([event.time])
-    fig3.canvas.draw()  # update the figure
+    fig5.canvas.draw()  # update the figure
 
 
+# Setup the events for the curstom plot. Moving the mouse will trigger a
+# matplotlib event, which we will respond to by publishing an MNE-Python UI
+# event. Upon receiving a UI event, we will move the vertical line.
 plt.connect("motion_notify_event", on_motion_notify)
-subscribe(fig3, "time_change", on_time_change)
+subscribe(fig5, "time_change", on_time_change)
 
-# Link the new figure with the topomap plots, so that the TimeChange events are
-# sent to all of them.
-link(fig3, fig1)
-link(fig3, fig2)
+# Link all the figures together.
+link(fig3, fig4)
+link(fig3, fig5)
+link(fig4, fig5)
```

Comment:
Maybe for some future PR, but this makes me think the API should be:
```
def link(*figures, ...):
```
Then this could just be
```suggestion
link(fig3, fig4, fig5)
```

Reply:
great idea!